R/llik.R

```diff
@@ -0,0 +1,151 @@
+#' Calculate log-likelihood
+#'
+#' @description
+#' Calculates the log-likelihood of a set of cross-sectional antibody response data, for a given incidence rate (`lambda`) value.
+#'
+#' @param pop_data a [data.frame()] with cross-sectional serology data per antibody and age, and additional columns
+#' @param antigen_isos Character vector listing one or more antigen isotypes. Values must match `pop_data`.
+#' @param curve_params a [data.frame()] containing MCMC samples of parameters from the Bayesian posterior distribution of a longitudinal decay curve model. The parameter columns must be named:
+#' - `antigen_iso`: a [character()] vector indicating antigen-isotype combinations
+#' - `iter`: an [integer()] vector indicating MCMC sampling iterations
+#' - `y0`: baseline antibody level at $t=0$ ($y(t=0)$)
+#' - `y1`: antibody peak level (ELISA units)
+#' - `t1`: duration of infection
+#' - `alpha`: antibody decay rate (1/days for the current longitudinal parameter sets)
+#' - `r`: shape factor of antibody decay
+#' @param noise_params a [data.frame()] (or [tibble::tibble()]) containing the following variables, specifying noise parameters for each antigen isotype:
+#' * `antigen_iso`: antigen isotype whose noise parameters are being specified on each row
+#' * `nu`: biological noise
+#' * `eps`: measurement noise
+#' * `y.low`: lower limit of detection for the current antigen isotype
+#' * `y.high`: upper limit of detection for the current antigen isotype
+#'
+#' @param verbose logical: if TRUE, print verbose log information to console
+#' @param ... additional arguments passed to other functions (not currently used).
+#' @inheritParams fdev
+#' @return the log-likelihood of the data with the current parameter values
+#' @export
+#' @examples
+#' library(dplyr)
+#' library(tibble)
+#'
+#' #load in longitudinal parameters
+#' dmcmc = load_curve_params("https://osf.io/download/rtw5k")
+#'
+#'
+#' #Specify the antibody-isotype responses to include in analyses
+#' antibodies = c("HlyE_IgA", "HlyE_IgG")
+#'
+#'  #set seed to reproduce results
+#' set.seed(54321)
+#'
+#' # simulated incidence rate per person-year
+#' lambda <- 0.2;
+#'
+#' # range covered in simulations
+#' lifespan <- c(0, 10);
+#'
+#' # cross-sectional sample size
+#' nrep <- 100
+#'
+#' # biologic noise distribution
+#' dlims <- rbind(
+#'   "HlyE_IgA" = c(min = 0, max = 0.5),
+#'   "HlyE_IgG" = c(min = 0, max = 0.5))
+#'
+#'
+#' #generate cross-sectional data
```

Comment:
@kristinawlai Please rework this example to use data from OSF instead of simulated data. I think it will be more concise and understandable that way.